platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinBuildScriptModelBuilder.kt

```diff
@@ -63,10 +63,19 @@ import java.util.EnumSet
 
 
 internal
-data class KotlinBuildScriptModelParameter(
-    val scriptFile: File?,
+interface KotlinBuildScriptModelParameter {
+    val scriptFile: File?
     val correlationId: String?
-)
+    val sourceSetClassPath: SourceSetClassPath?
+}
+
+
+internal
+data class DefaultKotlinBuildScriptModelParameter(
+    override val scriptFile: File?,
+    override val correlationId: String?,
+    override val sourceSetClassPath: SourceSetClassPath? = null,
```

Comment:
💭 It's a bit unclear for me, what is exactly `SourceSetClassPath` is represents?

Reply:
Ah, it seems like it's a classpath used to compile given sourceset

Reply:
Your question made me think about it more, and I found a solution that does not require the `SourceSetClassPath` at all, effectively keeping it more similar to the existing code.

Thanks for raising a question 👍